### PR TITLE
Users/suaggar/dynamic extension list

### PR DESCRIPTION
**Pull request by SumiranAgg**
Dynamic install extensions drop down

Issue to remove escape character in extension title:
https://github.com/toby-meyer/AzSecurityPackRedirectAndHeadersMed/issues/4

### CODE REVIEW

File: Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts [hunk 1]
```diff
@@ -45,17 +45,29 @@ export class KuduServiceUtils {
...
         }
+        for(var siteExtension of allSiteExtensions) {
+            allSiteExtensionMap[siteExtension.id] = siteExtension;
+        }
 
         for(var extensionID of extensionList) {
             var siteExtensionDetails = null;
-        
-             // Python extensions are moved to Nuget and the extensions IDs are changed. The belo check ensures that old extensions are mapped to new extension ID.
+            if(!allSiteExtensionMap[extensionID]){
```
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts]: nit: space before {

File: Tasks/Common/azure-arm-rest/Tests/azure-arm-app-service-kudu-tests.ts [hunk 2]
```diff
@@ -129,6 +129,26 @@ export class KuduTests {
         }
     }
 
+    public static async getAllSiteExtensions() {
```
Comment (vincentdass) [inline Tasks/Common/azure-arm-rest/Tests/azure-arm-app-service-kudu-tests.ts]: 👍 for L0

File: Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts [hunk 3]
```diff
@@ -45,17 +45,29 @@ export class KuduServiceUtils {
...
+        var allSiteExtensions = await this._appServiceKuduService.getAllSiteExtensions();
         var anyExtensionInstalled: boolean = false;
         var siteExtensionMap = {};
+        var allSiteExtensionMap = {};
         var extensionLocalPaths: string = "";
         for(var siteExtension of siteExtensions) {
             siteExtensionMap[siteExtension.id] = siteExtension;
         }
+        for(var siteExtension of allSiteExtensions) {
+            allSiteExtensionMap[siteExtension.id] = siteExtension;
```
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts]: suggestion: Can we include the title too in the map?

Comment (SumiranAgg) [reply Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts]: added

File: Tasks/Common/azure-arm-rest/Tests/azure-arm-app-service-kudu-tests.ts (hunk 2 above)
Comment (SumiranAgg) [reply Tasks/Common/azure-arm-rest/Tests/azure-arm-app-service-kudu-tests.ts]: Thanks :)

File: Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts (hunk 1 above)
Comment (SumiranAgg) [reply Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts]: ok

File: Tasks/AzureAppServiceManage/Strings/resources.resjson/en-US/resources.resjson [hunk 4]
```diff
@@ -141,6 +141,7 @@
   "loc.messages.FailedToStartContinuousWebJob": "Failed to start continuous WebJob '%s'. Error: %s",
   "loc.messages.FailedToStopContinuousWebJob": "Failed to stop continuous WebJob '%s'. Error: %s",
   "loc.messages.FailedToGetSiteExtensions": "Failed to get site extensions. Error: %s",
+  "loc.messages.FailedToGetAllSiteExtensions": "Failed to get all site extensions. Error: %s",
```
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/Strings/resources.resjson/en-US/resources.resjson]: nit: Failed to get extension feed

File: Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts [hunk 5]
```diff
@@ -45,17 +45,25 @@ export class KuduServiceUtils {
...
+            allSiteExtensionMap[siteExtension.title] = siteExtension;
+        }
 
         for(var extensionID of extensionList) {
             var siteExtensionDetails = null;
-        
-             // Python extensions are moved to Nuget and the extensions IDs are changed. The belo check ensures that old extensions are mapped to new extension ID.
+            if(allSiteExtensionMap[extensionID] && allSiteExtensionMap[extensionID].title == extensionID) {
```
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts]: (if **not** allSiteExtensionMap[extensionID]) ?

File: Tasks/AzureAppServiceManage/Strings/resources.resjson/en-US/resources.resjson (hunk 4 above)
Comment (SumiranAgg) [reply Tasks/AzureAppServiceManage/Strings/resources.resjson/en-US/resources.resjson]: incorporated

File: Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts (hunk 5 above)
Comment (SumiranAgg) [reply Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts]: we are updating extensionID only when its value is matching title of some siteextension.

**Comment by vincentdass** [comment]
#7036

**Pull request merged**